libmount/src/tab_parse.c

```diff
@@ -323,7 +323,14 @@ static int mnt_parse_utab_line(struct libmnt_fs *fs, const char *s)
 		if (!*p)
 			break;
 
-		if (!fs->id && !strncmp(p, "ID=", 3)) {
+		if (!fs->uniq_id && !strncmp(p, "UNIQID=", 7)) {
+			int rc = 0;
+
+			end = next_u64(p + 3, &fs->uniq_id, &rc);
```

Comment:
I have fixed the +7 issue. 

It would be possible to rewrite all the parser by startswith() to avoid all the hardcoded offset as startswith() returns a pointer after the prefix:
```C
if (!fs->uniq_id && (x = startswith(p, "UNIQID="))) {
    end = next_u64(x, &fs->uniq_id, &rc);
```
But it would be better to have separate PR for this. 

It would also be possible to introduce startsnwith() to accept the length (to avoid strlen()) of the prefix and add a new macro startwithconst():
```C
#define startswithconst(_s, _prefix)    startsnwith(_s, _prefix, sizeof(_prefix) - 1)
```
I guess we have many places where startswith() is used for constant prefixes.

Reply:
AFAIK the compiler can already inlline the existing function as its an inline function and it can constant-fold `strlen(CONSTANT)`.
My testing seems to confirm this.